test/solanaEncode.test.ts

```diff
@@ -8,11 +30,37 @@ describe("solanaEncode", () => {
       gateway: "94U5AHQMKkV5txNJ17QPXWoh474PheGou6cNP2FEuL1d",
     };
 
-    const expected =
-      "0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000000030b650b5c456bbc1c19bb901bb1fe3538d91402596d1675db7e3d40fc2f98306a0000000000000000000000000000000000000000000000000000000000000001803e2c72756c303bd21a5b950e10ab343e82118c123dbf6e53e326c496649b4d0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003736f6c0000000000000000000000000000000000000000000000000000000000";
+    const connectedProgramId = new anchor.web3.PublicKey(input.connected);
+    const [connectedPdaAccount] = anchor.web3.PublicKey.findProgramAddressSync(
+      [Buffer.from("connected", "utf-8")],
+      connectedProgramId
+    );
+    const [gatewayPdaAccount] = anchor.web3.PublicKey.findProgramAddressSync(
+      [Buffer.from("meta", "utf-8")],
+      new anchor.web3.PublicKey(input.gateway)
+    );
 
     const result = await solanaEncode(input);
-    expect(result).toBe(expected);
+    const { accounts, data: encodedData } = decodeEncodedResult(result);
+
+    expect(encodedData).toBe(ethers.hexlify(ethers.toUtf8Bytes(input.data)));
+    expect(accounts).toHaveLength(4);
+
+    expect(accounts[0].publicKey).toBe(hexlifyPubkey(connectedPdaAccount));
+    expect(accounts[0].isWritable).toBe(true);
+
+    expect(accounts[1].publicKey).toBe(hexlifyPubkey(gatewayPdaAccount));
+    expect(accounts[1].isWritable).toBe(false);
+
+    expect(accounts[2].publicKey).toBe(
+      hexlifyPubkey(anchor.web3.SystemProgram.programId)
+    );
+    expect(accounts[2].isWritable).toBe(false);
+
+    expect(accounts[3].publicKey).toBe(
+      hexlifyPubkey(anchor.web3.SYSVAR_INSTRUCTIONS_PUBKEY)
+    );
+    expect(accounts[3].isWritable).toBe(false);
```

Comment:
should we also add some `remainingAccounts`?

Reply:
Not sure I understand what you mean 🤔